sdk/identity/Azure.Identity/src/DefaultAzureCredential.cs

```diff
@@ -155,6 +162,8 @@ private static async ValueTask<(AccessToken, TokenCredential)> GetTokenFromSourc
                         ? await sources[i].GetTokenAsync(requestContext, cancellationToken).ConfigureAwait(false)
                         : sources[i].GetToken(requestContext, cancellationToken);
 
+                    _credentialType = sources[i].GetType();
```

Comment:
I believe this member assignment will break the thread safety of this class. We can't assign the cached credential and a member variable non-atomically.  Perhaps we can change the CredentialType property to get the type of the cached credential if it exists instead?

Reply:
After discussing offline, I moved the location of the assignment to mitigate the issue.